lib/py/src/transport/TSocket.py

```diff
@@ -151,7 +151,8 @@ def open(self):
     def read(self, sz):
         try:
             buff = self.handle.recv(sz)
-        except socket.timeout as e:
+        # TODO: remove socket.timeout when 3.10 becomes the earliest version of python supported.
+        except (socket.timeout, TimeoutError) as e:
             raise TTransportException(type=TTransportException.TIMED_OUT, message="read timeout", inner=e)
         except socket.error as e:
```

Comment:
similarly `socket.error` is also going to be a deprecated alias to [`OSError`](https://docs.python.org/3.12/library/exceptions.html#OSError) so we probably want to do something similar here.